src/server/shared_thread_local.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <memory>
+
+#include "server/engine_shard_set.h"
+#include "util/proactor_pool.h"
+
+namespace dfly {
+
+// Allows duplicating some class T to all threads via thread-local storage.
+// Calls to Set() will copy the shared_ptr<T> to thread_local variables, which can then be quickly
+// accessed without locks or atomic operations locally.
+// This class is thread-safe, but is not copyable nor movable.
+template <typename T> class SharedThreadLocal {
+ public:
+  static bool HasValue() {
+    return tl_t_ != nullptr;
+  }
+
+  static T* Get() {
+    return tl_t_.get();
+  }
+
+  static void Set(T t) {
+    Set(std::make_shared<T>(std::move(t)));
+  }
+
+  static void Set(std::shared_ptr<T> t) {
+    auto cb = [&](util::ProactorBase* pb) { tl_t_ = t; };
+    shard_set->pool()->AwaitFiberOnAll(std::move(cb));
+  }
+
+ private:
+  static thread_local std::shared_ptr<T> tl_t_;
+};
+
+template <typename T> thread_local std::shared_ptr<T> SharedThreadLocal<T>::tl_t_;
```

Comment:
I am not sure what is the value of this class, as we use define here thread_local shard_ptr , we dont have any other place in code that uses  thread_local shard_ptr except cluster family. Specificly in cluster family we wanted to use a shared ptr as we needed it to be on heap because the struct was to big and also shared because you want to build the data once (also validate once) on config update and than update all the pointer in all threads. I dont see any need for this class right now